kaspresso/src/main/kotlin/com/kaspersky/kaspresso/interceptors/behavior/impl/autoscroll/FallbackAutoScrollToAction.kt

```diff
@@ -0,0 +1,91 @@
+package com.kaspersky.kaspresso.interceptors.behavior.impl.autoscroll
+
+import android.util.Log
+import android.view.View
+import android.view.ViewParent
+import android.widget.HorizontalScrollView
+import android.widget.ListView
+import android.widget.ScrollView
+import androidx.core.widget.NestedScrollView
+import androidx.test.espresso.PerformException
+import androidx.test.espresso.UiController
+import androidx.test.espresso.ViewAction
+import androidx.test.espresso.matcher.ViewMatchers
+import androidx.test.espresso.util.HumanReadables
+import org.hamcrest.Matcher
+import org.hamcrest.Matchers
+
+class FallbackAutoScrollToAction : ViewAction {
+    override fun getConstraints(): Matcher<View> {
+        return Matchers.allOf(
+            ViewMatchers.withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE),
+            ViewMatchers.isDescendantOfA(
+                Matchers.anyOf(
+                    ViewMatchers.isAssignableFrom(NestedScrollView::class.java),
+                    ViewMatchers.isAssignableFrom(ScrollView::class.java),
+                    ViewMatchers.isAssignableFrom(HorizontalScrollView::class.java),
+                    ViewMatchers.isAssignableFrom(ListView::class.java)
+                )
+            )
+        )
+    }
+
+    private fun View?.isScrollable(): Boolean =
+        (this is ScrollView || this is NestedScrollView || this is HorizontalScrollView || this is ListView)
+
+    private tailrec fun View?.findFirstParentScrollableView(lastView: View): View? {
+        return if (this == lastView) {
+            if (this.isScrollable()) this else null
+        } else {
+            if (this.isScrollable()){
+                this
+            } else {
+                this?.parent?.findFirstView()?.findFirstParentScrollableView(lastView)
+            }
+        }
+    }
+
+    private tailrec fun ViewParent?.findFirstView(): View? {
+        return if (this is View) {
+            this
+        } else {
+            this?.parent.findFirstView()
+        }
+    }
+
+    override fun perform(uiController: UiController, view: View) {
+        if (ViewMatchers.isDisplayingAtLeast(100).matches(view)) {
+            Log.i(TAG, "View is already displayed. Returning.")
```

Comment:
It's better to use `logger: UiTestLogger`. 
See `AutoScrollFallbackProviderImpl` constructor.

Reply:
I followed suit from the ScrollToAction from Kakao, which uses the Log.i
If we pass the UiTestLogger we make it more Kaspresso-dependent, and I am still not sure whether this is something that should come here or in Kakao, which provides most ViewActions

Reply:
@RuslanMingaliev Need third opinion here =)

Reply:
@RuslanMingaliev @eakurnikov  ping

Reply:
> I think it makes sense to test this with Kautomator and Compose. I believe this is not necessary for Kautomator due to the way it scrolls (depends on the size of the screen). For compose, I am ver unsure.
> 
> I'll get this checked whenever I find time in the next 2 weeks
> 
> * [x]  Interceptor for Kautomator (not needed!)
> * [-]  Interceptor for Compose?

Since compose is something experimental, and this would likely require an additional compose solution, I will leave it out of scope for this.
In case there is an issue we should create another ticket :)

Reply:
If we will filter logs by KASPRESSO tag as we do always when debugging our tests, we will not see this log. I would say it is better to use single logger for all Kaspresso operations.

Reply:
@eakurnikov I will do it then

Reply:
- [ ] Add KASPRESSO filter logs